### PR TITLE
[Fluid] Embedded solvers unification

### PR DESCRIPTION
**Pull request by rubenzorrilla**
@jcotela As we've discussed so many times, this PR unifies all the embedded solvers in the main one. The old cases as well as all the tests are still compatible so there shouldn't be any problem when merging this (hopefully...).

### CODE REVIEW

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py
Comment (philbucher) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: wouldn't it make more sense to import this only if you need it?
Otherwise this will be a restriction for users

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py
Comment (philbucher) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: I think it would be cleaner to have a solver deriving from this which does the fm_ale stuff

you are mixing up things which I think makes it harder to understand / follow the code

File: applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_embedded_solver.py
Comment (philbucher) [inline applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_embedded_solver.py]: does the trilinos version work with the fm_ale?

**Review by philbucher — COMMENTED** [review]
some comments

File: applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_embedded_solver.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_embedded_solver.py]: We need to further enhance it to make it compatible (MPI search utility and probably add some calls to the communicator). We have plans to do it soon.

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: This is what we had before this PR and we (I'm speaking in behalf of @jcotela)  decided to remove it. The point is that such fm_ale solver has barely difference with the "common" one. We both think that it doesn't pay the price to have an alternative one (the FM-ALE util is deactivated by default). Indeed, any moving boundaries problem should be accompanied by the FM-ALE utility,

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: This check is already included in the Python solvers wrapper.

https://github.com/KratosMultiphysics/Kratos/blob/cd65b1a96bc72383416dabeb27dd93284f7cb52b/applications/FluidDynamicsApplication/python_scripts/python_solvers_wrapper_fluid.py#L68-L69

Comment (philbucher) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: but with what you have now you cannot use the embedded solver unless you have MehsMoving compiled
=> my point is not the check, but the unconditional import of MeshMoving
Or do you need it in any case?

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py
Comment (philbucher) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: ok

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: That's true... I only need it in case the FM-ALE is activated. Maybe is better to import it on the fly.

Comment (philbucher) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: yep
Take a look at the output of the tests...

**Review by jcotela — COMMENTED** [review]
Overall I agree with the changes. Regarding the import of the Meshing application, an alternative would be to import it conditionally at the beginning, as we do for tests, and set a flag if it failed. Then we can throw an error message later, only if the specific functionality from the meshing application is required.

My only review comment would be to ask you to take a look at how the stabilized formulation is chosen in the monolithic solver (the following json block and the code that reads it):
https://github.com/KratosMultiphysics/Kratos/blob/6ee8014726b646dab0bd0509028017d49820d4a1/applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py#L117-L119

and see if we can use a similar approach to choose between the alternative embedded formulations. I believe it would make sense to swap the `formulation` and `stabilization` labels of the base class if we are going to reuse it here (I volunteer to do it if you think it is a worthwhile change) and "package" all method details on a `formulation` sub-block (dynamic_tau, penalty_factor, ...)

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py]: :man_facepalming:  Solved!

**Comment by rubenzorrilla** [comment]
@jcotela @philbucher now I think it should do the job. @jginternational we need to slightly modify the GiD interface .json writing.

**Comment by jcotela** [comment]
@rubenzorrilla I've pushed the changes to use the same format you have here in the monolithic fluid solvers. I think it is a healthy change, but it makes this PR an API-breaker (or at least API-deprecator, since the old format still works)

**Comment by philbucher** [comment]
Guys could you please also update the GiD-Interface at some point soon?
That would be very much appreciated :)

**Comment by jcotela** [comment]
@philbucher It is on the pipeline, but we wanted to have a "finalized" format before messing with the interface. Also, I think we need to update the process list processing in the fluid scripts, too...

**Comment by rubenzorrilla** [comment]
@jginternational we have some duties for tomorrow.

**Comment by rubenzorrilla** [comment]
@philbucher can you approve? We won't merge until the GiD interface is ready.

File: applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_solver_vmsmonolithic.py
Comment (philbucher) [inline applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_solver_vmsmonolithic.py:75]: +1 for unification :+1:

**Review by philbucher — APPROVED** [review]
If you want you can go ahead with merging

**Comment by rubenzorrilla** [comment]
Merging, the interface is supposed to work once PR https://github.com/KratosMultiphysics/GiDInterface/pull/489 is merged (I want @jginternational to have a look to it before merging).

**Pull request merged**
